src/js/services/trackCopyCode.js

```diff
@@ -0,0 +1,22 @@
+const trackCopyCode = () => {
+  const copyBtnArr = Array.from(
+    document.getElementsByClassName('copy-to-clipboard-button'),
+  );
+
+  copyBtnArr.forEach((btn, index) => {
+    btn.addEventListener('click', () => {
+      const closestTocSection =
+        document.getElementById('active-toc-section').textContent;
```

Comment:
added `id` attribute to `toc.js` where we are adding the `active` class to help give some context to pages where there are a lot of snippets. This can help provide reference for around where the snippet is on the page in addition to having `codeSnippetPosition`

Reply:
Smart idea!